_stbt/grid.py

```diff
@@ -20,106 +22,179 @@ class Grid(object):
+    def get(self, index=None, position=None, region=None, data=None):
+        """Retrieve a single cell in the Grid.
 
-        For example in this grid "I" is index ``9`` and position
-        ``(x=3, y=1)``::
+        For example, let's say that you're looking for the selected item in
+        a grid by matching a reference image of the selection border. Then you
+        can find the (x, y) position in the grid of the selection, like this::
 
-            ABCDE
-            FGHIJ
-            KLMNO
+            selection = stbt.match("selection.png")
+            cell = grid.get(region=selection.region)
+            position = cell.position
 
-        :param int index: A 1D index into the grid, starting from 0 at the top
-            left, counting along the top row left to right, then the next row,
-            etc.
-            A negative index counts backwards from the end of the grid (so
-            ``-1`` is the bottom right position).
+        You must specify one (and only one) of ``index``, ``position``,
+        ``region``, or ``data``. For the meaning of these parameters see
+        `Grid.Cell`. A negative index counts backwards from the end of the grid
+        (so ``-1`` is the bottom right position).
 
-        :rtype: Position
-        :returns: x & y indexes into the grid (zero-based).
+        :returns: The `Grid.Cell` that matches the specified query; raises
+            `IndexError` if the index/position/region is out of bounds or the
+            data is not found.
         """
+        if len([x for x in [index, position, region, data]
+                if x is not None]) != 1:
+            raise ValueError("Exactly one of index, position, region, or data "
+                             "must be specified")
+        if data is not None and self.data is None:
+            raise IndexError("Searching by data %r but this Grid doesn't have "
+                             "any data associated" % data)
+        if index is not None:
+            position = self._index_to_position(index)
+            region = (None if self.region is None
+                      else self._position_to_region(position))
+        elif position is not None:
+            index = self._position_to_index(position)
+            region = (None if self.region is None
+                      else self._position_to_region(position))
+        elif region is not None:
+            position = self._region_to_position(region)
+            index = self._position_to_index(position)
+        elif data is not None:
+            for i in range(self.cols * self.rows):
+                position = self._index_to_position(i)
+                if data == self.data[position[1]][position[0]]:
+                    index = i
+                    region = (None if self.region is None
+                              else self._position_to_region(position))
+                    break
+            else:
+                raise IndexError("data '%r' not found" % data)
```

Comment:
I'm not sure if you want quoting around `%r`, if it's a str it will end up double quoted.  Also: `data` could be a `tuple` in which case this will fail.

```suggestion
                raise IndexError("data %r not found" % (data,))
```